### ISSUE TITLE
Devices not connecting to WiFi

### ISSUE THREAD

**Issue by skos-ninja** [DoESLiverpool/somebody-should#826]
TLDR: The USG (router) seemed to not be providing DHCP leases.

I'm not sure why this was happening but after a lot of attempts we fixed this by restarting a lot of things we found the route cause to be the router not providing DHCP leases. I couldn't figure out why but the usual solution of turning it off and on again seemed to fix this.

Both @amcewen and @johnmckerrell have the UniFi logins so can restart the router if need be. If no one is around you can just pull the power out of the device and back in however this should be a last resort.

**Comment by amcewen** [comment]
Although there were periods last night when this was fixed, I don't *think* it's resolved now.

**Comment by johnmckerrell** [comment]
This seems to still be an issue, we can't get the doorbots to reliably connect at the moment (could also be an issue with the doorbots but tricky to diagnose at the moment). I tried changing the WPA encryption from `AES/CCMP Only` to `Auto` and it seemed like that helped but some of the doorbots still aren't connecting. If it is that though the following link might be of use:

https://www.pi-point.co.uk/configs-hostapd/wpa-2-encryption-aes-ccmp/

**Comment by johnmckerrell** [comment]
I tried disabling advanced features, force provisioning each device and rebooting one of the Pi devices but it still didn't come up. I'm hoping to use ethernet to at least send out an updated config sometime today and then hopefully @skos-ninja can fix it fully tonight.

**Comment by ajlennon** [comment]
fwiw I seemed to be finding that my RPis could get onto the network but couldn't seem to pick up an IP address. If it will help I can make a date to sit with somebody with one of my PIs set up with a screen and Raspian and such and check through it getting onto the network?

**Comment by johnmckerrell** [comment]
Have you seen it recently? Over Wi-Fi, Ethernet or both?

**Comment by ajlennon** [comment]
Not recently. The last time I was working with a RPi was a week or two ago.

It did not successfully connect over WiFi but it did successfully connect over a wired connection.

This was running a vanilla Raspian.

**Comment by ajlennon** [comment]
I have a bag of five Pis that have been lent to me which are on my desk which we can plug into some displays and check

**Comment by ajlennon** [comment]
Bit more info: I've got a Raspberry Pi 3 B V2 here which won't connect to DoeS Liverpool at all. It will connect to Tapestry Free fine though. I've tried a couple of different operating system builds on it and both have the same problem. It can't get onto the AP at all

**Comment by skos-ninja** [comment]
Hi @ajlennon,

I have made a DoES Testing network with the password thisisatestnetwork.
Would you be able to try connecting that Pi to the network and leaving it connected.

It would be handy if I could have the mac address of that Pi to monitor it connecting.

**Comment by ajlennon** [comment]
`iwlist wlan0 scan`

  Cell 11 - Address: 02:9F:C2:F4:1B:E4
                    Channel:6
                    Frequency:2.437 GHz (Channel 6)
                    Quality=59/70  Signal level=-51 dBm  
                    Encryption key:on
                    ESSID:"DoES Testing"
                    Bit Rates:1 Mb/s; 2 Mb/s; 5.5 Mb/s; 11 Mb/s; 6 Mb/s
                              9 Mb/s; 12 Mb/s; 18 Mb/s
                    Bit Rates:24 Mb/s; 36 Mb/s; 48 Mb/s; 54 Mb/s
                    Mode:Master
                    Extra:tsf=0000000000000000
                    Extra: Last beacon: 11860ms ago
                    IE: Unknown: 000C446F45532054657374696E67
                    IE: Unknown: 010882848B968C129824
                    IE: Unknown: 030106
                    IE: Unknown: 2A0100
                    IE: Unknown: 3204B048606C
                    IE: Unknown: 0B0500004F0000
                    IE: Unknown: 2D1AAC011BFFFFFF0000000000000000000100000000000000000000
                    IE: Unknown: 3D1606080C00000000000000000000000000000000000000
                    IE: Unknown: 7F080000000000000040
                    IE: Unknown: DD180050F2020101000003A4000027A4000042435E0062322F00
                    IE: Unknown: DD0900037F01010000FF7F
                    IE: Unknown: DD1300156D00010100010227E58106F09FC2F31BE4
                    IE: IEEE 802.11i/WPA2 Version 1
                        Group Cipher : CCMP
                        Pairwise Ciphers (1) : CCMP
                        Authentication Suites (1) : PSK

**Comment by ajlennon** [comment]
`wpa_supplicant.conf`

trl_interface=DIR=/var/run/wpa_supplicant GROUP=netdev
update_config=1

network={
        ssid="DoES Testing"
        psk="this..."
        key_mgmt=WPA-PSK
}

**Comment by ajlennon** [comment]
`pi@raspberrypi:/etc/wpa_supplicant $ sudo wpa_supplicant -i wlan0 -c /etc/wpa_supplicant/wpa_supplicant.conf -B -d`

wpa_supplicant v2.4
random: Trying to read entropy from /dev/random
Successfully initialized wpa_supplicant
Initializing interface 'wlan0' conf '/etc/wpa_supplicant/wpa_supplicant.conf' driver 'default' ctrl_interface 'N/A' bridge 'N/A'
Configuration file '/etc/wpa_supplicant/wpa_supplicant.conf' -> '/etc/wpa_supplicant/wpa_supplicant.conf'
Reading configuration file '/etc/wpa_supplicant/wpa_supplicant.conf'
ctrl_interface='DIR=/var/run/wpa_supplicant GROUP=netdev'
update_config=1
Priority group 0
   id=0 ssid='DoES Testing'
rfkill: initial event: idx=0 type=1 op=0 soft=0 hard=0
rfkill: initial event: idx=1 type=2 op=0 soft=0 hard=0
nl80211: Supported cipher 00-0f-ac:1
nl80211: Supported cipher 00-0f-ac:5
nl80211: Supported cipher 00-0f-ac:2
nl80211: Supported cipher 00-0f-ac:4
nl80211: Using driver-based off-channel TX
nl80211: Supported vendor command: vendor_id=0x1018 subcmd=1
nl80211: Use separate P2P group interface (driver advertised support)
nl80211: Enable multi-channel concurrent (driver advertised support)
nl80211: use P2P_DEVICE support
nl80211: interface wlan0 in phy phy0
nl80211: Set mode ifindex 3 iftype 2 (STATION)
nl80211: Subscribe to mgmt frames with non-AP handle 0xc3d8a0
nl80211: Register frame type=0xd0 (WLAN_FC_STYPE_ACTION) nl_handle=0xc3d8a0 match=040a
nl80211: Register frame type=0xd0 (WLAN_FC_STYPE_ACTION) nl_handle=0xc3d8a0 match=040b
nl80211: Register frame type=0xd0 (WLAN_FC_STYPE_ACTION) nl_handle=0xc3d8a0 match=040c
nl80211: Register frame type=0xd0 (WLAN_FC_STYPE_ACTION) nl_handle=0xc3d8a0 match=040d
nl80211: Register frame type=0xd0 (WLAN_FC_STYPE_ACTION) nl_handle=0xc3d8a0 match=090a
nl80211: Register frame type=0xd0 (WLAN_FC_STYPE_ACTION) nl_handle=0xc3d8a0 match=090b
nl80211: Register frame type=0xd0 (WLAN_FC_STYPE_ACTION) nl_handle=0xc3d8a0 match=090c
nl80211: Register frame type=0xd0 (WLAN_FC_STYPE_ACTION) nl_handle=0xc3d8a0 match=090d
nl80211: Register frame type=0xd0 (WLAN_FC_STYPE_ACTION) nl_handle=0xc3d8a0 match=0409506f9a09
nl80211: Register frame type=0xd0 (WLAN_FC_STYPE_ACTION) nl_handle=0xc3d8a0 match=7f506f9a09
nl80211: Register frame type=0xd0 (WLAN_FC_STYPE_ACTION) nl_handle=0xc3d8a0 match=0801
nl80211: Register frame type=0xd0 (WLAN_FC_STYPE_ACTION) nl_handle=0xc3d8a0 match=06
nl80211: Register frame type=0xd0 (WLAN_FC_STYPE_ACTION) nl_handle=0xc3d8a0 match=0a07
nl80211: Register frame type=0xd0 (WLAN_FC_STYPE_ACTION) nl_handle=0xc3d8a0 match=0a11
nl80211: Register frame type=0xd0 (WLAN_FC_STYPE_ACTION) nl_handle=0xc3d8a0 match=1101
nl80211: Register frame type=0xd0 (WLAN_FC_STYPE_ACTION) nl_handle=0xc3d8a0 match=1102
nl80211: Register frame type=0xd0 (WLAN_FC_STYPE_ACTION) nl_handle=0xc3d8a0 match=0505
netlink: Operstate: ifindex=3 linkmode=1 (userspace-control), operstate=5 (IF_OPER_DORMANT)
nl80211: driver param='(null)'
Add interface wlan0 to a new radio phy0
nl80211: Regulatory information - country=00
nl80211: 2402-2472 @ 40 MHz 20 mBm
nl80211: 2457-2482 @ 20 MHz 20 mBm (no IR)
nl80211: 2474-2494 @ 20 MHz 20 mBm (no OFDM) (no IR)
nl80211: 5170-5250 @ 80 MHz 20 mBm (no IR)
nl80211: 5250-5330 @ 80 MHz 20 mBm (DFS) (no IR)
nl80211: 5490-5730 @ 160 MHz 20 mBm (DFS) (no IR)
nl80211: 5735-5835 @ 80 MHz 20 mBm (no IR)
nl80211: 57240-63720 @ 2160 MHz 0 mBm
nl80211: Added 802.11b mode based on 802.11g information
wlan0: Own MAC address: b8:27:eb:4b:84:25
wpa_driver_nl80211_set_key: ifindex=3 (wlan0) alg=0 addr=(nil) key_idx=0 set_tx=0 seq_len=0 key_len=0
wpa_driver_nl80211_set_key: ifindex=3 (wlan0) alg=0 addr=(nil) key_idx=1 set_tx=0 seq_len=0 key_len=0
wpa_driver_nl80211_set_key: ifindex=3 (wlan0) alg=0 addr=(nil) key_idx=2 set_tx=0 seq_len=0 key_len=0
wpa_driver_nl80211_set_key: ifindex=3 (wlan0) alg=0 addr=(nil) key_idx=3 set_tx=0 seq_len=0 key_len=0
wpa_driver_nl80211_set_key: ifindex=3 (wlan0) alg=0 addr=(nil) key_idx=4 set_tx=0 seq_len=0 key_len=0
wpa_driver_nl80211_set_key: ifindex=3 (wlan0) alg=0 addr=(nil) key_idx=5 set_tx=0 seq_len=0 key_len=0
wlan0: RSN: flushing PMKID list in the driver
nl80211: Flush PMKIDs
TDLS: TDLS operation not supported by driver
TDLS: Driver uses internal link setup
TDLS: Driver does not support TDLS channel switching
wlan0: WPS: UUID based on MAC address: 8bc60c95-c7f9-52d1-b48b-13685217bec2
ENGINE: Loading dynamic engine
ENGINE: Loading dynamic engine
EAPOL: SUPP_PAE entering state DISCONNECTED
EAPOL: Supplicant port status: Unauthorized
nl80211: Skip set_supp_port(unauthorized) while not associated
EAPOL: KEY_RX entering state NO_KEY_RECEIVE
EAPOL: SUPP_BE entering state INITIALIZE
EAP: EAP entering state DISABLED
ctrl_interface_group=108 (from group name 'netdev')
wlan0: Added interface wlan0
wlan0: State: DISCONNECTED -> DISCONNECTED
nl80211: Set wlan0 operstate 0->0 (DORMANT)
netlink: Operstate: ifindex=3 linkmode=-1 (no change), operstate=5 (IF_OPER_DORMANT)
nl80211: Create interface iftype 10 (P2P_DEVICE)
nl80211: New P2P Device interface p2p-dev-wlan0 (0x3) created
Initializing interface 'p2p-dev-wlan0' conf '/etc/wpa_supplicant/wpa_supplicant.conf' driver 'nl80211' ctrl_interface 'DIR=/var/run/wpa_supplicant GROUP=netdev' bridge 'N/A'
Configuration file '/etc/wpa_supplicant/wpa_supplicant.conf' -> '/etc/wpa_supplicant/wpa_supplicant.conf'
Reading configuration file '/etc/wpa_supplicant/wpa_supplicant.conf'
ctrl_interface='DIR=/var/run/wpa_supplicant GROUP=netdev'
update_config=1
Priority group 0
   id=0 ssid='DoES Testing'
rfkill: initial event: idx=0 type=1 op=0 soft=0 hard=0
rfkill: initial event: idx=1 type=2 op=0 soft=0 hard=0
nl80211: Supported cipher 00-0f-ac:1
nl80211: Supported cipher 00-0f-ac:5
nl80211: Supported cipher 00-0f-ac:2
nl80211: Supported cipher 00-0f-ac:4
nl80211: Using driver-based off-channel TX
nl80211: Supported vendor command: vendor_id=0x1018 subcmd=1
nl80211: Use separate P2P group interface (driver advertised support)
nl80211: Enable multi-channel concurrent (driver advertised support)
nl80211: use P2P_DEVICE support
nl80211: interface p2p-dev-wlan0 in phy phy0
nl80211: Set mode ifindex 0 iftype 10 (P2P_DEVICE)
nl80211: Failed to set interface 0 to mode 10: -22 (Invalid argument)
nl80211: Subscribe to mgmt frames with non-AP handle 0xc505a0
nl80211: Register frame type=0xd0 (WLAN_FC_STYPE_ACTION) nl_handle=0xc505a0 match=040a
nl80211: Register frame type=0xd0 (WLAN_FC_STYPE_ACTION) nl_handle=0xc505a0 match=040b
nl80211: Register frame type=0xd0 (WLAN_FC_STYPE_ACTION) nl_handle=0xc505a0 match=040c
nl80211: Register frame type=0xd0 (WLAN_FC_STYPE_ACTION) nl_handle=0xc505a0 match=040d
nl80211: Register frame type=0xd0 (WLAN_FC_STYPE_ACTION) nl_handle=0xc505a0 match=090a
nl80211: Register frame type=0xd0 (WLAN_FC_STYPE_ACTION) nl_handle=0xc505a0 match=090b
nl80211: Register frame type=0xd0 (WLAN_FC_STYPE_ACTION) nl_handle=0xc505a0 match=090c
nl80211: Register frame type=0xd0 (WLAN_FC_STYPE_ACTION) nl_handle=0xc505a0 match=090d
nl80211: Register frame type=0xd0 (WLAN_FC_STYPE_ACTION) nl_handle=0xc505a0 match=0409506f9a09
nl80211: Register frame type=0xd0 (WLAN_FC_STYPE_ACTION) nl_handle=0xc505a0 match=7f506f9a09
nl80211: Register frame type=0xd0 (WLAN_FC_STYPE_ACTION) nl_handle=0xc505a0 match=0801
nl80211: Register frame type=0xd0 (WLAN_FC_STYPE_ACTION) nl_handle=0xc505a0 match=06
nl80211: Register frame type=0xd0 (WLAN_FC_STYPE_ACTION) nl_handle=0xc505a0 match=0a07
nl80211: Register frame type=0xd0 (WLAN_FC_STYPE_ACTION) nl_handle=0xc505a0 match=0a11
nl80211: Register frame type=0xd0 (WLAN_FC_STYPE_ACTION) nl_handle=0xc505a0 match=1101
nl80211: Register frame type=0xd0 (WLAN_FC_STYPE_ACTION) nl_handle=0xc505a0 match=1102
nl80211: Register frame type=0xd0 (WLAN_FC_STYPE_ACTION) nl_handle=0xc505a0 match=0505
nl80211: Start P2P Device p2p-dev-wlan0 (0x3): Success
nl80211: driver param='(null)'
Add interface p2p-dev-wlan0 to existing radio phy0
nl80211: Regulatory information - country=00
nl80211: 2402-2472 @ 40 MHz 20 mBm
nl80211: 2457-2482 @ 20 MHz 20 mBm (no IR)
nl80211: 2474-2494 @ 20 MHz 20 mBm (no OFDM) (no IR)
nl80211: 5170-5250 @ 80 MHz 20 mBm (no IR)
nl80211: 5250-5330 @ 80 MHz 20 mBm (DFS) (no IR)
nl80211: 5490-5730 @ 160 MHz 20 mBm (DFS) (no IR)
nl80211: 5735-5835 @ 80 MHz 20 mBm (no IR)
nl80211: 57240-63720 @ 2160 MHz 0 mBm
nl80211: Added 802.11b mode based on 802.11g information
p2p-dev-wlan0: Own MAC address: c6:04:de:85:92:39
p2p-dev-wlan0: RSN: flushing PMKID list in the driver
nl80211: Flush PMKIDs
p2p-dev-wlan0: State: DISCONNECTED -> INACTIVE
p2p-dev-wlan0: WPS: UUID from the first interface: 8bc60c95-c7f9-52d1-b48b-13685217bec2
ENGINE: Loading dynamic engine
ENGINE: Loading dynamic engine
EAPOL: SUPP_PAE entering state DISCONNECTED
EAPOL: Supplicant port status: Unauthorized
nl80211: Skip set_supp_port(unauthorized) while not associated
EAPOL: KEY_RX entering state NO_KEY_RECEIVE
EAPOL: SUPP_BE entering state INITIALIZE
EAP: EAP entering state DISABLED
Using existing control interface directory.
ctrl_interface_group=108 (from group name 'netdev')
P2P: Add operating class 81
P2P: Channels - hexdump(len=13): 01 02 03 04 05 06 07 08 09 0a 0b 0c 0d
P2P: Own listen channel: 81:1
P2P: Random operating channel: 81:1
P2P: initialized
P2P: channels: 81:1,2,3,4,5,6,7,8,9,10,11,12,13
P2P: cli_channels:
p2p-dev-wlan0: Added interface p2p-dev-wlan0
p2p-dev-wlan0: State: INACTIVE -> DISCONNECTED
nl80211: Set p2p-dev-wlan0 operstate 0->0 (DORMANT)
netlink: Operstate: ifindex=0 linkmode=-1 (no change), operstate=5 (IF_OPER_DORMANT)
Daemonize..

**Comment by DoESsean** [comment]
Vikki has come in to use the laser this afternoon, and her laptop wouldn't work on our network - it was there, but just had the little yellow warning triangle when she connected to it.

She's got onto the free Fabric District wifi, but it's not ideal.

**Comment by skos-ninja** [comment]
So after much debugging tonight I can confirm this issue with Pi's not connecting is due to some of the older generations only supporting WPA1 or defaulting to it. However the network had WPA1 disabled.

I have now enabled WPA1 even though it has known vulnerabilities but this will only affect the client and not our network so I'm happy to leave this enabled for now.

**Comment by ajlennon** [comment]
That's really odd. OK I will test my Pi out... Ta

**Comment by ajlennon** [comment]
Nope still not working

I would be very surprised if this is WPA1 related as I'm using a stock Raspbian image here for test, just downloaded, and I'm imagining it's not going to be only supporting WPA1

**Comment by skos-ninja** [comment]
Well as the network is set to broadcast on both 5GHz and 2.4Ghz this shouldn't be an issue

**Comment by ajlennon** [comment]
e.g

https://www.raspberrypi.org/forums/viewtopic.php?t=192541

**Comment by ajlennon** [comment]
If it's not a 2.4 vs 5 issue then it's clear that one model of Pi works and one model of Pi doesn't so we should be able to get to the bottom of other differences between them. My money would still be on a setting related to 2.4 vs 5 atm

**Comment by ajlennon** [comment]
Hi @magman2112 @skos-ninja - I'm trying to use a wired connection in Room 29 instead of using the wifi. I tried the bank of 4 RJ45 connectors on the far wall (opposite the entry door) but there's no connection there. Is there something we could easily plug in to make one of them work? Thx !

**Comment by skos-ninja** [comment]
These are not patched in so no not currently

**Comment by ajlennon** [comment]
how big a job is that?

**Comment by skos-ninja** [comment]
As cables are patched in by @magman2112 and he has a layout defined for it I would leave it to him to do.

**Comment by ajlennon** [comment]
OK thanks - can you comment on what's involved @magman2112 ? Thanks

**Comment by magman2112** [comment]
Showed @ajlennon how to patch connections in Room 29, so hard wired connections now working in Room 2.

**Comment by skos-ninja** [comment]
At last we seem to have found the fix for this.

Turns out UniFi has a wonderful bug that means if you disable seamless roaming it doesn't actually disable and as such this seems to have been causing issues on older devices that do not support it as UniFi was telling the device to move to a different AP and the devices did not understand it properly and as such disconnecting and reconnecting constantly.

I fixed this by restoring from an old backup of the UniFi config that never had it enabled. I have now made a copy of this config and asked for it to be placed in the somewhere-private repo.

Let's hope this is the last attempt at getting this fixed. 🎉
